File: Task 21.py
def getMissedElement(arr):
    maxElement = max(arr)
    totalSum = sum(range(1, maxElement + 1))
    return totalSum - sum(arr)
    
data = {
    (1, 3, 4, 5) : 2,
    (1, 2, 3, 4, 6, 7) : 5,
    (2, 3, 4) : 1
    } 

for arr in data:
    if getMissedElement(arr) == data[arr]:
        print('Answer for {0} is {1}'.format(arr, getMissedElement(arr)))
    else:
        print('Text failed')
        break
else:
    print('Text successful')

File: Task 21.cs
using System;

public class Programm
{
	public static int GetMaxElement(int[] arr)
	{
		int max = arr[0];
		foreach (int i in arr)
		{
			if (i > max)
			{
				max = i;
			}
		}
		return max;
	}
	
	public static int SumOfRange(int N)
	{
		int sum = 0;
		for (int i = 1; i < N + 1; i++)
		{
			sum += i;
		}
		return sum;
	}
	
	
	public static int GetMissedElement(int[] arr)
	{
		int max = GetMaxElement(arr);
		int fullSum = SumOfRange(max);
		int sum = 0;
		foreach (int num in arr)
		{
			sum += num;
		}
		return fullSum - sum;
	}
	
	public static void Main()
	{
		Console.WriteLine("Testing of GetMissedElement");
		int[] data = new int[] {1, 2, 3, 5};
		Console.WriteLine("Answer for {0} is {1}", data, GetMissedElement(data));
		
	}
}
